api/dataprocessor.go

```diff
@@ -124,119 +129,123 @@ func (s *Server) getTargets(ctx context.Context, reqs []models.Req) ([]models.Se
 		}
 	}
 
-	var mu sync.Mutex
 	var wg sync.WaitGroup
 
-	out := make([]models.Series, 0)
-	errs := make([]error, 0)
-
+	responses := make(chan getTargetsResp, 1)
+	getCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
https://godoc.org/golang.org/x/net/context#WithCancel

cancel should always be called otherwise resources will leak.  `go vet` will warn if you dont call cancel with something like
```
api/dataprocessor.go:135: the cancel function is not used on all paths (possible context leak)
```